### PR TITLE
ISPN-14171 Support RFC 6598 shared address space addresses

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14171

### CODE REVIEW

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/MultiHomedServerAddress.java [hunk 1]
```diff
@@ -145,15 +145,23 @@ public InetAddressWithNetMask(InetAddress address, short prefixLength, boolean n
...
                } else if (a[0] == (byte) 172 && a[1] >= 16 && a[1] <= 31) {
+                  // RFC 1918 Private network 172.16.0.0/12
                   this.prefixLength = 12;
                } else if (a[0] == (byte) 169 && a[1] == (byte) 254) {
+                  // RFC 3927 Link-local addresses 169.254.0.0/16
                   this.prefixLength = 16;
                } else if (a[0] >= (byte) 240 && a[0] <= (byte) 255) {
+                  // RFC 1112 Reserved network 240.0.0.0/4
                   this.prefixLength = 4;
+               } else if (a[0] == (byte) 100 && a[1] >= (byte) 64) {
```
Comment (jabolina) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/MultiHomedServerAddress.java:162]:
```suggestion
               } else if (a[0] == (byte) 100 && a[1] >= (byte) 64 && a[1] <= (byte) 127) {
```

Comment (tristantarrant) [reply server/hotrod/src/main/java/org/infinispan/server/hotrod/MultiHomedServerAddress.java:162]: image

**Comment by tristantarrant** [comment]
@jabolina I've changed the code so that it is much more extensible and will also support custom ranges in the future

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Merged, thanks @tristantarrant

**Pull request merged**